pkgs/by-name/do/dovecot_fts_xapian/generic.nix

```
@@ -0,0 +1,67 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  autoreconfHook,
+  sqlite,
+  pkg-config,
+  xapian,
+  icu,
+  ...
+}:
+{
+  # fts-xapian 1.9 is compatible with dovecot 2.2, 2.3 & 2.4
+  # so no need for per-version hashes
+  dovecot,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "dovecot-fts-xapian";
+  version = "1.9.1";
+
+  src = fetchFromGitHub {
+    owner = "grosjo";
+    repo = "fts-xapian";
+    rev = version;
+    hash = "sha256-+8THyzzBV8QQVQFeKCSvIzkr5oaE0vdWU2gsolChfoo=";
+  };
+
+  buildInputs = [
+    xapian
+    icu
+    sqlite
+  ];
+
+  nativeBuildInputs = [
+    autoreconfHook
+    pkg-config
+  ];
+
+  # patching dovecot_moduledir in ${dovecot}/lib/dovecot/dovecot-config to point to $out/lib/dovecot/modules instead of ${dovecot}/lib/dovecot/modules
+  postPatch = ''
+    mkdir dovecot
+    sed "s&dovecot_moduledir=${dovecot}/lib/dovecot/modules&dovecot_moduledir=${placeholder "out"}/lib/dovecot/modules&" ${dovecot}/lib/dovecot/dovecot-config > dovecot/dovecot-config
```

Use substituteInPlace

---

this one doesn't substitute in place, it redirects the `sed` output to a new file that we then use in `configureFlags`

---

putting this here so I don't forget: `"--with-dovecot=/build/source/dovecot"` can probably be replaced with `"--with-dovecot=$BUILD_DIR/dovecot"`

---

> this one doesn't substitute in place, it redirects the `sed` output to a new file that we then use in `configureFlags`

I missed that. Maybe we can use cp and then substituteInPlace? I think we really want to abort the build if the replacement fails instead of quietly continuing

---

didn't know that sed would just continue, good idea
I'll fix it later ^^